### PR TITLE
Changed the "umalloc.c" format to the caffeinix format

### PR DESCRIPTION
**Pull request by GoKo-Son626**
#### One doubt

I found that on line 78 of the original file, the corresponding code in xv6 is
```
if((prevp = freep) == 0){
```
But it's
```
if((prevp = freep) == &base){
```
That's not the same, so I question that.

### CODE REVIEW

File: mkfs/mkfs.c [hunk 1]
```diff
@@ -44,260 +44,269 @@ void iappend(uint inum, void *p, int n);
...
+        char buf[BSIZE];
+        struct dinode din;
+
+
+        static_assert(sizeof(int) == 4, "Integers must be 4 bytes!");
+
+        if(argc < 2){
+        fprintf(stderr, "Usage: mkfs fs.img files...\n");
+                exit(1);
+        }
```
Comment (TroyMitchell911) [inline mkfs/mkfs.c]: Require space between parentheses and curly braces in 82 line.
Like this:
```c
if(argc < 2) {
```

File: mkfs/mkfs.c [hunk 2]
```diff
@@ -44,260 +44,269 @@ void iappend(uint inum, void *p, int n);
...
+                // get rid of "user/"
+                char *shortname = strstr(argv[i], "user/");
+
+                if(!shortname)
+                {
+                        shortname = argv[i];
+                } else 
+                {
+                        shortname += 5;
+                }
```
Comment (TroyMitchell911) [inline mkfs/mkfs.c]: Wrong placement of brackets.

File: mkfs/mkfs.c [hunk 3]
```diff
@@ -44,260 +44,269 @@ void iappend(uint inum, void *p, int n);
...
+        uchar buf[BSIZE];
+        int i;
+
+        printf("balloc_mkfs: first %d blocks have been allocated\n", used);
+        assert(used < BSIZE*8);
+        bzero_mkfs(buf, BSIZE);
+        for(i = 0; i < used; i++)
+        {
+                buf[i/8] = buf[i/8] | (0x1 << (i%8));
+        }
```
Comment (TroyMitchell911) [inline mkfs/mkfs.c]: Wrong placement of brackets

File: mkfs/mkfs.c [hunk 4]
```diff
@@ -44,260 +44,269 @@ void iappend(uint inum, void *p, int n);
...
+                                din.addrs[NDIRECT] = xint(freeblock++);
+                        }
+                        rsect(xint(din.addrs[NDIRECT]), (char*)indirect);
+                        if(indirect[fbn - NDIRECT] == 0)
+                        {
+                                indirect[fbn - NDIRECT] = xint(freeblock++);
+                                wsect(xint(din.addrs[NDIRECT]), (char*)indirect);
+                        }
+                        x = xint(indirect[fbn-NDIRECT]);
+                }
```
Comment (TroyMitchell911) [inline mkfs/mkfs.c]: Wrong placement of brackets

File: user/umalloc.c [hunk 5]
```diff
@@ -11,88 +11,96 @@
...
+        {
+                bp->s.size += p->s.ptr->s.size;
+                bp->s.ptr = p->s.ptr->s.ptr;
+        } else
+                bp->s.ptr = p->s.ptr;      
+        if(p + p->s.size == bp)
+        {
+                p->s.size += bp->s.size;
+                p->s.ptr = bp->s.ptr;
+        } else
```
Comment (TroyMitchell911) [inline user/umalloc.c]: Wrong placement of brackets

File: user/umalloc.c [hunk 6]
```diff
@@ -11,88 +11,96 @@
...
+                                p += p->s.size;
+                                p->s.size = nunits;
+                        }
+                        freep = prevp;
+                        return (void*)(p + 1);
+                }
+                if(p == freep)
+                        if((p = morecore(nunits)) == 0)
+                                return 0;
+        }
```
Comment (TroyMitchell911) [inline user/umalloc.c]: Wrong placement of brackets

**Comment by TroyMitchell911** [comment]
> #### One doubt
> I found that on line 78 of the original file, the corresponding code in xv6 is
> 
> ```
> if((prevp = freep) == 0){
> ```
> 
>  Save
> But it's
> 
> ```
> if((prevp = freep) == &base){
> ```
> 
>  Save
> That's not the same, so I question that.

You can see the definition of `freep`.It was assigned the address of the variable `base` at the beginning.But in xv6, it is zero.
So that is why the `if` judgment is different

**Comment by GoKo-Son626** [comment]
OK, I known, Thanks

**Comment by GoKo-Son626** [comment]
Re-fix the format problem of "umalloc.c" and "umalloc.c" 
[Tested]implement some empty functions in ulib.c
[cace710]
(https://github.com/TroyMitchell911/caffeinix/pull/19/commits/cace7105ad76e62cb6c129f6106be64ecd5b3ac4)

**Comment by TroyMitchell911** [comment]
I have merged this PR! Thx!

**Pull request closed without merging**